new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItWITValidation.java

```diff
@@ -12,20 +12,28 @@
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import static oracle.weblogic.kubernetes.actions.ActionConstants.ARCHIVE_DIR;
 import static oracle.weblogic.kubernetes.actions.ActionConstants.MODEL_DIR;
 import static oracle.weblogic.kubernetes.actions.ActionConstants.WIT_BUILD_DIR;
+import static oracle.weblogic.kubernetes.actions.TestActions.buildAppArchive;
 import static oracle.weblogic.kubernetes.actions.TestActions.createMIIImage;
+import static oracle.weblogic.kubernetes.actions.TestActions.withAppParams;
 import static oracle.weblogic.kubernetes.actions.TestActions.withWITParams;
 import static oracle.weblogic.kubernetes.assertions.TestAssertions.dockerImageExists;
 import static oracle.weblogic.kubernetes.utils.FileUtils.checkDirectory;
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
 
 @DisplayName("Simple validation of basic WIT functions")
 class ItWITValidation implements LoggedTest {
   private static final String WDT_MODEL_FILE = "model1-wls.yaml";
   private static final String IMAGE_NAME = "test-mii-image-2";
   private static final String IMAGE_TAG = "v1";
   
+  // Name of the directory under resources/apps for an application
+  // Note: use "_" instead of "-" in app directories names because
+  // WDT does not like "-" to be in the name of an archive file
```

Comment:
@ddsharpe, is there a bug for this?  Generally, I don't want to obscure problems by having the tests workaround something that doesn't work.

Reply:
@rjeberhard it is not a known issue.  I just ran a quick test and do not see any issues.
./createDomain.sh -oracle_home ~/Oracle/Middleware12.2.1.3.0 -domain_parent . -model_file ~/Desktop/demo/my_domain.yaml -archive_file ~/Desktop/demo/my-domain.zip  -variable_file ~/Desktop/demo/my_domain1.10.properties

Reply:
I recall that before I changed "-" to "_" in the sourcePath in my model file, the following caused an error.
appDeployments:
    Application:
        myear:
            SourcePath: wlsdeploy/applications/sample_app.ear
            ModuleType: ear
            Target: 'cluster-1'

Reply:
Somehow WDT failed to parse the model yams file when the resourcePath contains "-". It seems that WDT considered -a as an option token and printed out this error:
####<Apr 10, 2020 6:00:40 PM> <INFO> <WebLogicDeployToolingVersion> <logVersionInfo> <WLSDPLY-01750> <The WebLogic Deploy Tooling validateModel version is 1.7.2:master.10eaf6e:Mar 20, 2020 23:24 UTC>
line 31:53 token recognition error at: '-a'
####<Apr 10, 2020 6:00:40 PM> <SEVERE> <ParsingErrorListener> <syntaxError> <WLSDPLY-18018> <Parse error for file /u01/wdt/models/model1-wls.yaml at line 31 position 55: extraneous input 'pp.ear' expecting {COMMENT, NEWLINE}>
####<Apr 10, 2020 6:00:40 PM> <SEVERE> <validate> <__perform_model_file_validation> <WLSDPLY-20009> <validateModel failed to load the model file /u01/wdt/models/model1-wls.yaml: Unable to parse model from file /u01/wdt/models/model1-wls.yaml : The YAML parser encountered 1 error(s) while parsing /u01/wdt/models/model1-wls.yaml>
####<Apr 10, 2020 6:00:40 PM> <SEVERE> <validate> <main> <WLSDPLY-20000> <validateModel encountered an unexpected validation error: Unable to parse model from file /u01/wdt/models/model1-wls.yaml : The YAML parser encountered 1 error(s) while parsing /u01/wdt/models/model1-wls.yaml>

Reply:
We could handle this by converting "-" to "_" in the AppBuilder action, but it is easier just not to have "-" in the app name.

Reply:
Once I added "" around the value of sourcePath, the "-" in the application ear file path worked.

Reply:
This comment is still here :).  Let's change "sample_app" to use "sample-app" just to prove that it is fixed.